### PR TITLE
Add parquet_ans_processor

### PR DESCRIPTION
**Pull request by yuunlimm**
### Description
- Migrate ans_primary_name_v2 over to BQ, which includes v1

### Test Plan
Screenshot 2024-07-11 at 11 04 15 AM
Screenshot 2024-07-11 at 11 04 12 AM

### CODE REVIEW

File: rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                {
+                    all_ans_primary_names_v2.push(ans_lookup_v2);
+                    transaction_version_to_struct_count
+                        .entry(txn_version)
+                        .and_modify(|e| *e += 1)
+                        .or_insert(1);
+                }
+            }
+
+            // Parse V1 ANS write set changes
```
[the remark this answers is not in the document]
Comment (yuunlimm) [reply rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs:200]: why do we have to parse V2 ans wsc?

File: rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    config: ParquetAnsProcessorConfig,
+    ans_primary_name_v2_sender: AsyncSender<ParquetDataGeneric<AnsPrimaryNameV2>>,
+}
+
+impl ParquetAnsProcessor {
+    pub fn new(
+        connection_pool: ArcDbPool,
+        config: ParquetAnsProcessorConfig,
+        new_gap_detector_sender: AsyncSender<ProcessingResult>,
+    ) -> Self {
```
Comment (larry-aptos) [inline rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs:60]: also `create_parquet_handler_loop` part is shared among processors.

File: rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        let all_ans_primary_names_v2 = parse_ans(
+            &transactions,
+            &mut transaction_version_to_struct_count,
+            self.config.ans_v1_primary_names_table_handle.clone(),
+            self.config.ans_v2_contract_address.clone(),
+        );
+
+        let ans_primary_name_v2_parquet_data = ParquetDataGeneric {
+            data: all_ans_primary_names_v2,
+            transaction_version_to_struct_count: transaction_version_to_struct_count.clone(),
```
Comment (larry-aptos) [inline rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs:119]: ...ParquetDataGeneric::default

File: rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs (hunk 2 above)
Comment (yuunlimm) [reply rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs:60]: sync offline, we will refactor this out to trait later

File: rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+pub struct AnsPrimaryNameV2 {
+    pub txn_version: i64,
+    pub write_set_change_index: i64,
+    pub registered_address: String,
+    pub token_standard: String,
+    pub domain: Option<String>,
+    pub subdomain: Option<String>,
+    pub token_name: Option<String>,
+    pub is_deleted: bool,
+    #[allocative(skip)]
```
Comment (yuunlimm) [inline rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs:36]: @ying-w let me know if it's missing any fields

Comment (ying-w) [reply rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs:36]: missing `expiration_timestamp` and `subdomain_expiration_policy`

Comment (ying-w) [reply rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs:36]: @rtso do you know how to parse out the expiration timestamps?

Comment (rtso) [reply rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs:36]: Primary names don't have expiration timestamp, only ans lookups do. This PR looks like it's missing moving ans_lookup_v2 to Parquet @yuunlimm

Comment (rtso) [reply rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs:36]: https://github.com/aptos-labs/aptos-indexer-processors/blob/e35189535ecce84eb654bc1fb3ea02ab72b6f52d/rust/processor/src/processors/ans_processor.rs#L737

Comment (rtso) [reply rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs:36]: I also realized that this processor is still writing to the v1 ANS tables 
```
all_current_ans_lookups,
all_ans_lookups,
all_current_ans_primary_names,
all_ans_primary_names,
```
We can include these in the deprecated tables and turn it off

Comment (yuunlimm) [reply rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs:36]: I don't think ans_lookup_v2 was a part of parquet migration.

Comment (yuunlimm) [reply rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs:36]: @ying-w do we need lookups in BQ?

Comment (ying-w) [reply rust/processor/src/db/common/models/ans_models/parquet_ans_lookup_v2.rs:36]: just realized this was for `ans_primary_name_v2` not `ans_lookup_v2`

Yes we use both `ans_lookup_v2` and `ans_primary_name_v2`

File: rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+            "AnsProcessor {{ connections: {:?}  idle_connections: {:?} }}",
+            state.connections, state.idle_connections
+        )
+    }
+}
+
+#[async_trait]
+impl ProcessorTrait for ParquetAnsProcessor {
+    fn name(&self) -> &'static str {
+        ProcessorName::AnsProcessor.into()
```
Comment (larry-aptos) [inline rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs:97]: do we want a processor name that can differentiate parquet and regular ones? I see this method is implemented in different ways: 1) from processor name 2) hard coding

**Review by larry-aptos — APPROVED** [review]

File: rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs (hunk 5 above)
Comment (yuunlimm) [reply rust/processor/src/processors/parquet_processors/parquet_ans_processor.rs:97]: good catch! will fix this.

**Pull request merged**